### DISCUSSION TITLE
How to Create a Design-Time Component Using Skia Canvas in Delphi Athens?

### DISCUSSION THREAD

**Discussion by Zeus64** [skia4delphi/skia4delphi#298] [Q&A]
I'm working on a custom component in Delphi Athens that utilizes the Skia Canvas for rendering. I've managed to get the component functioning correctly at runtime, but I'm encountering issues when trying to make it work at design time (AccessViolation). I'm wondering if I need to initialize something specifically for design time, such as setting ALGlobalUseSkia to true, but I'm unsure where and how to do this appropriately in a design-time context.

Any insights or examples would be greatly appreciated. Thank you in advance for your help!

**Comment by Zeus64** [comment]
I found, in design time, the main canvas is not a TSKCustomCanvas, end of the story :) I must use a "buffer" bitmap

**Reply by viniciusfbb** [reply]
Exactly. GlobalUseSkia is False by default, both at designtime and at runtime. That's why we created a base control called TSkCustomControl, to draw on a bitmap when the canvas is not Skia-based, but to draw directly on the canvas when it is Skia-based. This way, we allow users to easily enable/disable Skia rendering without affecting other library features, such as Skia controls. This helps a lot to check issues in the render, without many changes.

As for Skia rendering in designtime, we could later add an option in the IDE to enable GlobalUseSkia in designtime as well, because the rendering (canvas) based on Skia supports more features than the D2D canvas.
